packages/javascript-mutator/src/JavaScriptMutator.ts

```diff
@@ -17,39 +16,32 @@ export class JavaScriptMutator implements Mutator {
     const mutants: Mutant[] = [];
 
     inputFiles.forEach(file => {
+      const fileName = file.name;
+
       const ast = this.parser.parse(file.textContent);
 
       this.parser.getNodes(ast).forEach(node => {
         this.mutators.forEach(mutator => {
-          const mutatedNodes = mutator.mutate(node, copy);
-
-          if (mutatedNodes.length) {
-            const newMutants = this.generateMutants(mutatedNodes, mutator.name, file.name);
-            mutants.push(...newMutants);
-          }
+          const mutatorName = mutator.name;
+
+          mutator.mutate(node).forEach(([original, mutation]) => {
+            if (original.start !== null && original.end !== null) {
+              const replacement = types.isNode(mutation) ? this.parser.generateCode(mutation) : mutation.raw;
+
+              mutants.push({
+                fileName: file.name,
+                mutatorName: mutator.name,
```

Comment:
You have defined in line 25:
``const mutatorName = mutator.name``
isn't it better to write here:
``mutatorName,`` instead of ``mutatorName: mutator.name``

Reply:
Yes, and I made the same mistake for fileName above. I will fix these, thanks. I may need a few days to go through the other comments.

Reply:
should be fixed now